x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/TrainedModelAssignmentClusterService.java

```diff
@@ -811,7 +824,21 @@ private void updateNumberOfAllocations(
             listener.onFailure(ExceptionsHelper.missingModelDeployment(deploymentId));
             return;
         }
-        if (existingAssignment.getTaskParams().getNumberOfAllocations() == numberOfAllocations) {
+        AdaptiveAllocationsSettings adaptiveAllocationsSettings = getAdaptiveAllocationsSettings(
+            existingAssignment.getAdaptiveAllocationsSettings(),
+            adaptiveAllocationsSettingsUpdates
+        );
+        if (adaptiveAllocationsSettings != null) {
+            ActionRequestValidationException validationException = adaptiveAllocationsSettings.validate();
+            if (validationException != null) {
+                listener.onFailure(validationException);
+                return;
+            }
+        }
+        boolean hasUpdates = (numberOfAllocations != null
+            && Objects.equals(numberOfAllocations, existingAssignment.getTaskParams().getNumberOfAllocations()) == false)
+            || Objects.equals(adaptiveAllocationsSettings, existingAssignment.getAdaptiveAllocationsSettings()) == false;
```

Comment:
Could all knowledge of the adaptive allocations be removed from this class?

The scaler uses the adaptive allocation settings to make a decision about scaling returns a desired number of allocations. This cluster service just needs to know the scaler's decision in terms of the desired allocations

Reply:
I'm not sure what you're exactly saying.

The cluster service only has knowledge about the adaptive allocations _settings_ (enabled/min/max). The decision of the scaler isn't here.

Reply:
I was questioning why `adaptiveAllocationsSettingsUpdates ` is a parameter to the method at all as the logic to decide the number of allocations is done elsewhere. I now see that `adaptiveAllocationsSettingsUpdates` is passed into the `increase/decreaseNumberOfAllocations()` functions and the updated values stored in the `TrainedModelAssignmentMetadata`